network/notifee.go

```diff
@@ -1,90 +1,124 @@
-			time.Sleep(1 * time.Second)
-		}
+		for {
+			select {
+			case evt := <-s.lp2pEventSub.Out():
+				switch e := evt.(type) {
+				case lp2pevent.EvtLocalReachabilityChanged:
+					s.logger.Info("reachability changed", "reachability", e.Reachability)
 
-		if len(protocols) == 0 {
-			n.logger.Info("unable to get supported protocols", "pid", pid)
-		} else {
-			n.logger.Debug("get supported protocols", "pid", pid, "protocols", protocols)
-		}
+				case lp2pevent.EvtPeerConnectednessChanged:
+					s.logger.Debug("connectedness changed", "pid", e.Peer, "connectedness", e.Connectedness)
+
+				case lp2pevent.EvtPeerIdentificationCompleted:
+					s.logger.Debug("identification completed", "pid", e.Peer)
+					s.sendConnectEven(e.Peer)
+
+				case lp2pevent.EvtPeerProtocolsUpdated:
+					s.logger.Debug("protocols updated", "pid", e.Peer, "protocols", e.Added)
+					s.sendConnectEven(e.Peer)
 
-		n.peerMgr.AddPeer(pid, conn.RemoteMultiaddr(), conn.Stat().Direction, protocols)
+				default:
+					s.logger.Debug("unhandled libp2p event", "event", evt)
+				}
 
-		supportStream := slices.Contains(protocols, n.streamProtocolID)
-		n.eventChannel <- &ConnectEvent{
-			PeerID:        pid,
-			RemoteAddress: conn.RemoteMultiaddr().String(),
-			SupportStream: supportStream,
+			case <-s.ctx.Done():
+				return
+			}
 		}
 	}()
 }
 
-func (n *NotifeeService) Disconnected(_ lp2pnetwork.Network, conn lp2pnetwork.Conn) {
+func (s *NotifeeService) Stop() {
+}
+
+func (s *NotifeeService) Connected(_ lp2pnetwork.Network, conn lp2pnetwork.Conn) {
 	pid := conn.RemotePeer()
-	n.logger.Info("disconnected from peer", "pid", pid)
-	n.eventChannel <- &DisconnectEvent{PeerID: pid}
+	s.logger.Info("connected to peer", "pid", pid, "direction", conn.Stat().Direction)
+	s.peerMgr.AddPeer(pid, conn.RemoteMultiaddr(), conn.Stat().Direction)
+}
 
-	n.peerMgr.RemovePeer(pid)
+func (s *NotifeeService) Disconnected(_ lp2pnetwork.Network, conn lp2pnetwork.Conn) {
+	pid := conn.RemotePeer()
+	s.logger.Info("disconnected from peer", "pid", pid)
+	s.eventChannel <- &DisconnectEvent{PeerID: pid}
+
+	s.peerMgr.RemovePeer(pid)
 }
 
-func (n *NotifeeService) Listen(_ lp2pnetwork.Network, ma multiaddr.Multiaddr) {
+func (s *NotifeeService) Listen(_ lp2pnetwork.Network, ma multiaddr.Multiaddr) {
 	// Handle listen event if needed.
-	n.logger.Debug("notifee Listen event emitted", "addr", ma.String())
+	s.logger.Debug("notifee Listen event emitted", "addr", ma.String())
 }
 
 // ListenClose is called when your node stops listening on an address.
-func (n *NotifeeService) ListenClose(_ lp2pnetwork.Network, ma multiaddr.Multiaddr) {
+func (s *NotifeeService) ListenClose(_ lp2pnetwork.Network, ma multiaddr.Multiaddr) {
 	// Handle listen close event if needed.
-	n.logger.Debug("notifee ListenClose event emitted", "addr", ma.String())
+	s.logger.Debug("notifee ListenClose event emitted", "addr", ma.String())
+}
+
+func (s *NotifeeService) sendConnectEven(pid lp2pcore.PeerID) {
```

Comment:
```suggestion
func (s *NotifeeService) sendConnectEvent(pid lp2pcore.PeerID) {
```

just a small typo here.